catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/security/SystemTemplateAccessPlugin.java

```diff
@@ -0,0 +1,154 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.security;
+
+import static org.codice.ddf.catalog.ui.security.Constants.SYSTEM_TEMPLATE;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.Result;
+import ddf.catalog.operation.CreateRequest;
+import ddf.catalog.operation.DeleteRequest;
+import ddf.catalog.operation.DeleteResponse;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.operation.ResourceRequest;
+import ddf.catalog.operation.ResourceResponse;
+import ddf.catalog.operation.UpdateRequest;
+import ddf.catalog.plugin.AccessPlugin;
+import ddf.catalog.plugin.StopProcessingException;
+import java.io.Serializable;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import org.codice.ddf.catalog.ui.util.EndpointUtil;
+import org.codice.ddf.security.common.Security;
+
+public class SystemTemplateAccessPlugin implements AccessPlugin {
+  private final EndpointUtil util;
+
+  private final Supplier<Boolean> subjectHasAdmin;
+
+  public SystemTemplateAccessPlugin(EndpointUtil util) {
+    this(util, () -> Security.getInstance().javaSubjectHasAdminRole());
```

Comment:
This does work for a standard Java subject, but you should really be using the PDP to perform any sort of actual check. If you just build a policy instead, the FilterPlugin will just check against the PDP for you.